### PR TITLE
Add device suffix option for functional tests

### PR DESCRIPTION
**Pull request by tuxedcat**
### Tickets:
- 67491

### Previous PR
 - #13096

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/CMakeLists.txt [hunk 1]
```diff
@@ -26,7 +26,10 @@ endif()
 addIeTargetTest(
         NAME ${TARGET_NAME}
         ROOT ${CMAKE_CURRENT_SOURCE_DIR}
-        INCLUDES ${INCLUDES}
+        INCLUDES
+            ${INCLUDES}
+            ${CMAKE_SOURCE_DIR}/src/tests/functional/plugin/shared/include
```
Comment (iefode) [inline src/plugins/intel_cpu/tests/functional/CMakeLists.txt:31]: why did you make such changes in cmake? Common and shared are linked to the binary files

File: src/tests/functional/plugin/gpu/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,8 @@ addIeTargetTest(
             ${CMAKE_CURRENT_SOURCE_DIR}
         INCLUDES
             ${CMAKE_CURRENT_SOURCE_DIR}
+            ${CMAKE_SOURCE_DIR}/src/tests/functional/plugin/shared/include
+            ${CMAKE_SOURCE_DIR}/src/tests/ie_test_utils/common_test_utils
```
Comment (vladimir-paramuzov) [inline src/tests/functional/plugin/gpu/CMakeLists.txt:15]: Why do we need these extra includes?

File: src/tests/functional/plugin/gpu/shared_tests_instances/set_device_name.cpp [hunk 3]
```diff
@@ -5,23 +5,26 @@
...
-    auto available_devices = utils::PluginCache::get().core()->get_available_devices();
-    if (std::find(available_devices.begin(), available_devices.end(), new_gpu_name) == available_devices.end()) {
-        throw std::runtime_error("The device " + new_gpu_name + " is not in the available devices! Please use other on!");
-    }
+    static std::string new_gpu_name;
+    new_gpu_name =
+        std::string(CommonTestUtils::DEVICE_GPU) +
+        std::string(1, CommonTestUtils::DEVICE_SUFFIX_SEPARATOR) +
+        suffix;
+    // TODO: fix me (device suffix validation)
```
Comment (iefode) [inline src/tests/functional/plugin/gpu/shared_tests_instances/set_device_name.cpp:21]: Please uncomment! It works (have checked with @vladimir-paramuzov )

File: src/tests/functional/plugin/shared/include/set_device_name.hpp [hunk 4]
```diff
@@ -6,12 +6,12 @@
 
 #include <string>
 
-#include "common_test_utils/test_constants.hpp"
+#include "test_constants.hpp"
```
Comment (vladimir-paramuzov) [inline src/tests/functional/plugin/shared/include/set_device_name.hpp:9]: Seems like this change should be reverted and in that case cmake update won't be needed

**Review by vladimir-paramuzov — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/tests/functional/CMakeLists.txt (hunk 1 above)
Comment (tuxedcat) [reply src/plugins/intel_cpu/tests/functional/CMakeLists.txt]: Fixed.

File: src/tests/functional/plugin/gpu/shared_tests_instances/set_device_name.cpp (hunk 3 above)
Comment (tuxedcat) [reply src/tests/functional/plugin/gpu/shared_tests_instances/set_device_name.cpp]: Fixed.

File: src/tests/functional/plugin/gpu/CMakeLists.txt (hunk 2 above)
Comment (tuxedcat) [reply src/tests/functional/plugin/gpu/CMakeLists.txt]: Removed.

File: src/tests/functional/plugin/shared/include/set_device_name.hpp (hunk 4 above)
Comment (tuxedcat) [reply src/tests/functional/plugin/shared/include/set_device_name.hpp]: Fixed.

**Review by iefode — APPROVED** [review]
LGTM from ie test infra side

**Pull request merged**